test/common/network_policy_access_ns_to_svc.go

```diff
@@ -0,0 +1,264 @@
+	serviceName         = "testing-curl"
+	podName             = "testing-curl"
+	containerName       = "testing-curl"
+	threescaleNamespace = NamespacePrefix + "3scale"
+	podEndpoitResponse  = "success"
+)
+
+func TestNetworkPolicyAccessNSToSVC(t *testing.T, ctx *TestingContext) {
+
+	if err := createTestingIDP(goctx.TODO(), ctx.Client, http.DefaultClient); err != nil {
+		t.Fatalf("error while creating testing idp: %v", err)
+	}
+
+	rhmi, err := getRHMI(ctx.Client)
+	if err != nil {
+		t.Fatalf("error getting RHMI CR: %v", err)
+	}
+
+	// console master url
+	masterURL := rhmi.Spec.MasterURL
+
+	openshiftHTTPClient, err := resources.DoAuthOpenshiftUser(masterURL, "customer-admin-1", "Password1")
+	if err != nil {
+		t.Fatalf("error occured trying to get token : %v", err)
+	}
+
+	openshiftClient := &resources.OpenshiftClient{HTTPClient: openshiftHTTPClient}
+
+	// creating a project as dedicated-admin
+	_, err = createProject(ctx, masterURL, openshiftClient)
+	if err != nil {
+		t.Fatalf("error occured while creating a project: %v", err)
+	}
+
+	// creating service as dedicated-admin
+	serviceCR, err := createService(ctx, masterURL, openshiftClient)
+	if err != nil {
+		t.Fatalf("error occured while creating a service: %v", err)
+	}
+
+	// creating pod as dedicated-admin
+	podCR, err := createPodWithAnEndpoint(ctx, masterURL, openshiftClient)
+	if err != nil {
+		t.Fatalf("error occured while creating a pod: %v", err)
+	}
+
+	podReady, err := checkPodStatus(ctx, podCR)
+	if err != nil {
+		t.Fatalf("error checking pod status: %v", err)
+	}
+
+	if podReady == false {
+		t.Fatalf("pod %s failed to become ready", podCR.GetName())
+	}
+
+	tsApicastPod, err := get3ScaleAPIcastPod(ctx)
+	if err != nil {
+		t.Fatalf("error getting 3scale apicast pod: %v", err)
+	}
+
+	curlCommand := fmt.Sprintf("curl %s.%s.svc.cluster.local", serviceCR.GetName(), serviceCR.GetNamespace())
+	apicastContainerName := "apicast-production"
+	outputCurlCommand, err := execToPod(curlCommand,
+		tsApicastPod.GetName(),
+		tsApicastPod.GetNamespace(),
+		apicastContainerName,
+		ctx)
+	if err != nil {
+		t.Fatalf("error occured while executing curl command in pod: %v", err)
+	}
+
+	if strings.TrimSpace(outputCurlCommand) != podEndpoitResponse {
+		t.Fatalf("Failed to validate response Pod apicast received %s, instead of %s", outputCurlCommand, podEndpoitResponse)
+	}
+
+	cleanUp(ctx)
+
+}
+
+// creating project as dedicated admin
+func createProject(ctx *TestingContext, openshiftAPIURL string, openshiftClient *resources.OpenshiftClient) (*projectv1.ProjectRequest, error) {
+	projectCR := &projectv1.ProjectRequest{
+		ObjectMeta: v1.ObjectMeta{
+			Name: projectName,
+		},
+	}
+
+	// check if project exist
+	err := ctx.Client.Get(goctx.TODO(), k8sclient.ObjectKey{Name: projectName}, &projectv1.Project{})
+	if err != nil && !k8serr.IsNotFound(err) {
+		return nil, fmt.Errorf("error occured while retrieving a project: %v", err)
+	} else if k8serr.IsNotFound(err) {
```

Comment:
Probably don't need to explicitly declare the `else` statement. if there is an error and it is not an `isNotFound` error this if statement will be called anyway

Reply:
That makes sense but only if there is an error,  if there isn't any error `err` will be nil and then it'll attempt to create the CR even though it already exist. 
The reason for the check is to avoid return an error when the CR is present.